subprojects/base-services/src/main/java/org/gradle/internal/concurrent/DefaultParallelismConfiguration.java

```diff
@@ -21,12 +21,14 @@
 import org.gradle.concurrent.ParallelismConfiguration;
 
 import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class DefaultParallelismConfiguration implements Serializable, ParallelismConfiguration {
     public static final ParallelismConfiguration DEFAULT = new DefaultParallelismConfiguration();
 
     private boolean parallelProjectExecution;
     private int maxWorkerCount;
+    private final AtomicBoolean maxWorkerCountRetreived = new AtomicBoolean(false);
```

Comment:
This `AtomicBoolean` isn't getting you anything more than a `volatile boolean` would.  But given that the `maxWorkerCount` itself isn't `volatile` I'm not sure whether you'd want even that.

Reply:
Excellent point. Update to use `boolean`.